dashboard/lib/quick_assign_helper.rb

```diff
@@ -0,0 +1,36 @@
+module QuickAssignHelper
+  def self.course_offerings(user, locale)
+    offerings = {}
+
+    assignable_offerings = CourseOffering.assignable_course_offerings(user)
+    assignable_elementary_offerings = assignable_offerings.filter(&:elementary_school_level?)
+    assignable_middle_offerings = assignable_offerings.filter(&:middle_school_level?)
+    assignable_high_offerings = assignable_offerings.filter(&:high_school_level?)
+
+    offerings[:elementary] = group_offerings(assignable_elementary_offerings, user, locale)
+    offerings[:middle] = group_offerings(assignable_middle_offerings, user, locale)
+    offerings[:high] = group_offerings(assignable_high_offerings, user, locale)
+
+    offerings
+  end
+
+  def self.group_offerings(course_offerings, user, locale)
+    data = {}
+    course_offerings.each do |co|
+      next if co.header.blank? || co.curriculum_type.blank?
+
+      data[co.curriculum_type] ||= {}
+      data[co.curriculum_type][co.header] ||= []
+      data[co.curriculum_type][co.header].append(co.summarize_for_quick_assign(user, locale))
```

Comment:
This is all really great! I'm not all that familiar with ruby so selfishly I'd love some comments for what each of the sections do, but maybe that's just because it took me a minute to piece it all together:)

Reply:
Added some comments!